doc/api/deprecations.md

```diff
@@ -2617,6 +2617,22 @@ async function openAndClose() {
 }
 ```
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: `process.mainModule`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/32232
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only
+
+[`process.mainModule`][] is a CJS-only feature while `process` global object is
+shared with non-CJS environment.
+
+It's safe to use [`require.main`][] as a replacement.
```

Comment:
@jasnell mentioning the reason here if that's OK to you:
```suggestion
It is deprecated in favor of [`require.main`][], because it serves the same
purpose and is only available on CommonJS environment.
```